Yoast/YoastSEO.js
# 1947 improve keyphrase in title assessment

## Summary

This PR can be summarized in the following changelog entry:

* In the "keyphrase in title" assessment, function words preceding the exact match keyphrase are ignored when determining the position of the keyphrase in the title. 

## Relevant technical choices:

N/A

## Test instructions

This PR can be tested by following these steps:

* Start the webpack example in default mode (cd to `examples/webpack` and run `yarn start` in the terminal) and check that the behaviour of the assessment [did not change](https://github.com/Yoast/YoastSEO.js/wiki/Scoring-SEO-analysis#8-page-title-keyword-assessment).
* Start the webpack example in recalibration mode (cd to `examples/webpack` and run `yarn start-recalibration` in the terminal).
* In the **SEO Title** text input, enter (keyphrase = _Keyphrase research_):

| Description | Example |Feedback |
|:---------|:----------|:------|
| Keyphrase at the start of the title. | _Keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by one function word. | _Your keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by multiple function words. | _Your first keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by multiple function words plus a content word. | _Your sentence and first keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears in the SEO title, but not at the beginning. Try to move it to the beginning._ |
| Partial keyphrase in the title. | _Your sentence and first keyword research is awesome!_ | _Keyphrase in title: Not all the words from your keyphrase "Keyphrase research" appear in the SEO title. Try to use the exact match of your keyphrase in the SEO title._ |

* In the **SEO Title** text input, enter (keyphrase = _your keyphrase research_):

| Description | Example |Feedback |
|:---------|:----------|:------|
| Keyphrase at the start of the title. | _Your keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by one function word. | _And your keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by multiple function words. | _First off, your keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears at the beginning of the SEO title. Good job!_ |
| Keyphrase in the middle, preceded by multiple function words plus a content word. | _And sentence and your keyphrase research is awesome!_ | _Keyphrase in title: The exact match of the keyphrase appears in the SEO title, but not at the beginning. Try to move it to the beginning._ |
| Partial keyphrase in the title. | _Your sentence and first keyword research is awesome!_ | _Keyphrase in title: Not all the words from your keyphrase "Your keyphrase research" appear in the SEO title. Try to use the exact match of your keyphrase in the SEO title._ |

Fixes #1947
